src/python_testing/matter_testing_support.py

```diff
@@ -117,6 +121,76 @@ def get_default_paa_trust_store(root_path: pathlib.Path) -> pathlib.Path:
         return pathlib.Path.cwd()
 
 
+def parse_pics(lines=typing.List[str]) -> dict[str, bool]:
+    pics = {}
+    for raw in lines:
+        line, _, _ = raw.partition("#")
+        line = line.strip()
+
+        if line == "":
+            continue
+
+        key, _, val = line.partition("=")
+        val = val.strip()
+        if val not in ["1", "0"]:
+            raise ValueError('PICS {} must have a value of 0 or 1'.format(key))
+
+        pics[key.strip().upper()] = (val == "1")
+    return pics
+
+
+def read_pics_from_file(filename: str) -> dict[str, bool]:
+    """ Reads a dictionary of PICS from a file. """
+    with open(filename, 'r') as f:
+        lines = f.readlines()
+        return parse_pics(lines)
+
+
+def type_matches(received_value, desired_type):
+    """ Checks if the value recived matches the expected type.
+
+        Handles unpacking Nullable and Optional types and
+        compares list value types for non-empty lists.
+    """
+    if typing.get_origin(desired_type) == typing.Union:
+        for t in typing.get_args(desired_type):
+            if type_matches(received_value, t):
+                return True
+        return False
+    elif typing.get_origin(desired_type) == list:
+        # Assume an empty list is of the correct type
+        if isinstance(received_value, list):
+            if received_value == []:
+                return True
+            for t in typing.get_args(desired_type):
+                if type_matches(received_value[0], t):
```

Comment:
seems like `if foo: return True else return False` so maybe just

```py
return type_matches(received_value[0], t)
```

Reply:
never mind, did not notice indenting. Maybe some `any` loop works

Reply:
like:

```py
return any(typing.get_args(desired_type), lambda t: type_matches(received_value[0], t))
```

but this may be too much.

Reply:
Thoughts on the new way?